Add guide on what rate limit events are (#25252)
DataDog/documentation: content/en/metrics/guide/rate-limit.md

---
title: Events Generated from Rate Limited Metrics
private: true
further_reading:
- link: "/metrics/custom_metrics/"
  tag: "Documentation"
  text: "Custom Metrics"
- link: "/metrics/guide/custom_metrics_governance/"
  tag: "Documentation"
  text: "Best Practices for Custom Metrics Governance"
---

## Overview 

Submitting metrics to Datadog with large numbers of unique tag values on a given key can result in high [cardinality][1]. Most often, this is caused by unbounded tags.

> Unbounded and high cardinality tags can impact your account’s performance and stability. To protect your account, Datadog monitors metric increases and notifies you when these submissions are rate-limited.

Author: ```suggestion
Unbounded and high cardinality tags can impact your account's performance and stability. To protect your account, Datadog monitors metric increases and notifies you when these submissions are rate-limited.
```

{{< img src="/metrics/guide/rate_limit/event_monitor_config.png" alt="Event monitor configuration for a rate limit event" style="width:90%;" >}}

## Reduce metric cardinality

To remove the rate limit, you need to review the unbounded tags that are shown in the event. We recommend checking if all the tags are necessary and adjust the submission of the metric so that these tags are not present in such high cardinality.

For more information, see the [Best Practices for Custom Metrics Governance][4] guide.

## Submit a request to remove the rate limit

<div class="alert alert-warning">Only a Datadog Admin can request a removal of a metric rate limit. If you are not an Admin, make sure to include an Admin on the support ticket so they can confirm the request.</div>

After making the changes to remove the unbounded tags, submit a request to [Datadog Support][5] to remove the rate limit. In your request, provide the following information: 
- Name of the metric that was being rate limited 
- Link to the rate limit event in the Event Platform
- Confirmation a configuration change was done to remove the unbounded tags. 

> ## Further reading

Author: ```suggestion
## Further Reading
```

> - Name of the metric that was being rate limited 
> - Link to the rate limit event in the Event Platform
> - Confirmation a configuration change was done to remove the unbounded tags. 

Reviewer: ```suggestion
- Name of the rate-limited metric 
- Link to the rate limit event in the Event Platform
- Confirmation that a configuration change removed the unbounded tags
```
